src/util/saveArticles.ts

```diff
@@ -118,26 +118,62 @@ async function downloadBulk(listOfArguments: any[], downloader: Downloader): Pro
                     return resp;
                 });
             },
-            {concurrency: CONCURRENCY_LIMIT}
+            { concurrency: CONCURRENCY_LIMIT }
         );
     } catch (err) {
         logger.log(`Not able download in bulk due to ${err}`);
     }
 }
 
+async function getAllArticlesToKeep(downloader: Downloader, mw: MediaWiki, dump: Dump) {
+    if (!dump.customProcessor) {
+        return;
+    }
+    await articleDetailXId.iterateItems(
+        downloader.speed,
+        async (articleKeyValuePairs) => {
+            for (const [articleId] of Object.entries(articleKeyValuePairs)) {
+                try {
+                    const rets = await downloader.getArticle(articleId, dump);
+
+                    for (const { articleId, html: articleHtml } of rets) {
+                        if (!articleHtml) {
+                            continue;
+                        }
+
+                        const { articleDoc: _articleDoc } = await processArticleHtml(articleHtml, downloader, mw, dump, articleId);
```

Comment:
Why we do `processArticleHtml`, seems not necessary here to me.

Reply:
We need _articleDoc to perform operations on it. For eg: `doc.querySelector(`#fr.sectionlangue`);`
 `articleHtml` is a type string, we need to process it in doc to use methods on it.

Reply:
What you need is a DOM. To get the DOM from the HTML, you just need to call `let doc = domino.createDocument(html);` so why calling the most complex function of the whole MWoffliner instead of just calling that line?